File: gherkin/java/src/main/java/gherkin/ast/GherkinDocument.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public class GherkinDocument extends Node {
    private final Feature feature;
    private final List<Comment> comments;

    public GherkinDocument(
            Feature feature,
            List<Comment> comments) {
        super(null);
        this.feature = feature;
        this.comments = Collections.unmodifiableList(comments);
    }

    public Feature getFeature() {
        return feature;
    }

    public List<Comment> getComments() {
        return comments;
    }
}

File: gherkin/dotnet/Gherkin/Ast/GherkinDocument.cs
using System.Collections.Generic;

namespace Gherkin.Ast
{
    public class GherkinDocument
    {
        public Feature Feature { get; private set; }
        public IEnumerable<Comment> Comments { get; private set; }

        public GherkinDocument(Feature feature, Comment[] comments)
        {
            Feature = feature;
            Comments = comments;
        }
    }
}
